src/main.rs

```diff
@@ -13,13 +13,40 @@ use crate::inputs::{request_inputs, validate_inputs};
 use crate::trusted_dealer_keygen::trusted_dealer_keygen;
 
 fn main() -> io::Result<()> {
-    let config = request_inputs();
+    let mut reader = Box::new(io::stdin().lock());
+    let config = request_inputs(&mut reader);
+    match config {
+        Ok(_) => (),
+        Err(e) => {
+            eprintln!("Error: {}", e);
+            std::process::exit(exitcode::DATAERR)
+        }
+    }
+
+    let config = config.unwrap();
+
     let mut rng = thread_rng();
 
-    validate_inputs(&config).expect("An error occurred");
+    let valid = validate_inputs(&config);
+    match valid {
+        Ok(_) => (),
+        Err(e) => {
+            eprintln!("Error: {}", e);
+            std::process::exit(exitcode::DATAERR)
+        }
+    }
 
     // Print outputs
-    let (key_packages, pubkeys) = trusted_dealer_keygen(config, &mut rng);
+    let out = trusted_dealer_keygen(config, &mut rng);
+    match out {
+        Ok(_) => (),
+        Err(e) => {
+            eprintln!("Error: {}", e);
+            std::process::exit(1)
+        }
+    }
```

Comment:
Same as above

Reply:
Added in [ad3c315](https://github.com/ZcashFoundation/frost-zcash-demo/pull/4/commits/ad3c31588ba00a06aae1129cad967403e04c5fe7)